(DOCSP-16656) Backfills docs content for dbusers and events commands

sarahsimpers opened the pull request:
## Proposed changes

Backfills docs content for dbusers and events commands

_Jira ticket:_ 
https://jira.mongodb.org/browse/DOCSP-16656
https://jira.mongodb.org/browse/DOCSP-16657
https://jira.mongodb.org/browse/DOCSP-16658
https://jira.mongodb.org/browse/DOCSP-16659
https://jira.mongodb.org/browse/DOCSP-16661

## Checklist

- [ ] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added any necessary documentation in document requirements section listed in [CONTRIBUTING.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/CONTRIBUTING.md) (if appropriate)
- [ ] I have addressed the @mongodb/docs-cloud-team comments (if appropriate)
- [ ] I have updated [test/README.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/test/README.md) (if an e2e test has been added)
- [ ] I have run `make fmt` and formatted my code

gssbzn commented on internal/usage/usage.go:
```diff
@@ -34,7 +34,7 @@ const (
 	NIndexes                 = "Maximum number of indexes to suggest."
 	MDBVersion               = "Major MongoDB version of the cluster."
-	AuthDB                   = "Authentication database name."
+	AuthDB                   = "Authentication database name. If the user authenticates with AWS IAM, x.509, or LDAP, this value should be $external. If the user authenticates with SCRAM-SHA, this value should be admin."
```
this change affects ops manager/cloud manager docs as we are reusing the same docs and AWS IAM is not valid for those.
If you want to update this copy please create a new variable and use the new variable

gssbzn commented on internal/usage/usage.go:
```diff
@@ -34,7 +34,8 @@ const (
 	MDBVersion               = "Major MongoDB version of the cluster."
-	AuthDB                   = "Authentication database name."
+	AuthDB                   = "Authentication database name. If the user authenticates with AWS IAM, x.509, or LDAP, this value should be $external. If the user authenticates with SCRAM-SHA, this value should be admin."
+	MCLIAuthDB               = "Authentication database name."
```
it would've been easier to create `AtlasAuthDB` but if this is the way you want to play it please update to sue this new variable

https://github.com/mongodb/mongodb-atlas-cli/blob/73bd6d30c60771c02af52599b644aafaf323312f/internal/cli/opsmanager/dbusers/create.go#L131

sarahsimpers replied:
@gssbzn That makes sense, I can change it to do an Atlas variable as the exception. Right now I'm trying to figure out if I can make some examples only show up for Atlas CLI, so this PR may take a little time while I figure that out

sarahsimpers commented:
@gssbzn I removed the AWS examples altogether for now, may investigate later if I can add them in for only Atlas ones using annotations and editing the template. I also changed it to use a new variable for the Atlas CLI, not for MCLI. RFAL when you have time. Thank you!

kanchana-mongodb commented on internal/cli/atlas/dbusers/delete.go:
```diff
@@ -54,11 +55,16 @@ func DeleteBuilder() *cobra.Command {
 		Use:     "delete <username>",
 		Aliases: []string{"rm"},
-		Short:   "Delete a database user for your project.",
+		Short:   "Remove the specified database user from your project.",
```
To me, "your project" here sounds like command automatically detects project. Because user has to provide project ID also, should this be:
```suggestion
		Short:   "Remove the specified database user from the specified project.",
```

gssbzn replied:
there's a bit of nuance here, users may have set a default project, in which case the projectId flag is optional, adding this for context as I have no opinion on one copy being better than the other

sarahsimpers replied:
@kanchana-mongodb aside from Gus's comment, I wanted to avoid using "specified" twice per short description because it seems a little stiff and awkward. I do think it's important to say that this is the "specified user". So if you have any suggestions that are better than "your project" given that context, please let me know

kanchana-mongodb commented on internal/cli/atlas/dbusers/delete.go:
```diff
@@ -54,11 +55,16 @@ func DeleteBuilder() *cobra.Command {
 		Args:    require.ExactArgs(1),
 		Annotations: map[string]string{
-			"usernameDesc": "Username to delete from the MongoDB database.",
+			"usernameDesc": "Username to delete from the MongoDB database. The format of the username depends on the user's method of authentication.",
```
Question: Does this delete the user from the database or does this disassociate the user from the project? I am guessing the latter from the command and suggest the following: 
```suggestion
			"usernameDesc": "Username of user in the MongoDB database to delete from the specified project. The format of the username depends on the user's method of authentication.",
```

sarahsimpers replied:
I think that command would be atlas projects users delete and this does something different, but I will leave that for @gssbzn to answer

gssbzn replied:
yes, this is about clusters/deployments access and not project access if that makes it any clearer

gssbzn commented on internal/cli/atlas/dbusers/describe.go:
```diff
@@ -95,7 +92,11 @@ func DescribeBuilder() *cobra.Command {
 	}
 
-	cmd.Flags().StringVar(&opts.authDB, flag.AuthDB, convert.AdminDB, usage.AuthDB)
+	if config.BinName() == config.AtlasCLI {
+		cmd.Flags().StringVar(&opts.authDB, flag.AuthDB, convert.AdminDB, usage.AtlasAuthDB)
+	} else {
+		cmd.Flags().StringVar(&opts.authDB, flag.AuthDB, convert.AdminDB, usage.AuthDB)
+	}
```
there's no need for this `if` you can safety use `usage.AtlasAuthDB` with no major impact, the issue was with `usage.AuthDB` being reused for both ops manager and atlas commands, this is not about mongocli vs atlas cli

sarahsimpers replied:
Oh ok, I will change it

gssbzn commented on internal/cli/atlas/dbusers/describe.go:
```diff
@@ -62,20 +62,17 @@ func DescribeBuilder() *cobra.Command {
-  # The following example returns a MongoDB database user with AWS IAM authentication on an Atlas cluster. Prepend $external with \ to escape the special-use character.
-  %[1]s dbuser describe arn:aws:iam::772401394250:user/my-test-user --authDB \$external --output json`, cli.ExampleAtlasEntryPoint()),
+  # Return the details for the X.509-authenticating database user with the RFC 2253 Distinguished Name CN=ellen@example.com,OU=users,DC=example,DC=com. Prepend $external with \ to escape the special-use character:
+  %[1]s dbuser describe CN=ellen@example.com,OU=users,DC=example,DC=com --authDB \$external --output json`, cli.ExampleAtlasEntryPoint()),
```
you can safetly bring back the AWS IAM example, there's not problem with showing it for both atlas cli and mongocli, the only issue was with ops manager commands and only on the flags, examples are different

gssbzn reviewed (approved):
LGTM
